src/main/java/com/db/hackaton/repository/MedicalCaseRepository.java

```diff
@@ -25,4 +23,10 @@
     List<MedicalCase> findByRegistryIdAndCNP(@Param("registryId") Long registryId, @Param("cnp") String cnp);
 
     List<MedicalCase> findByStatusAndRegistryUuid(String status, String registryUuid);
+
+    @Query("select medicalCase from MedicalCase medicalCase JOIN Registry r ON r.uuid = medicalCase.registryUuid " +
```

Comment:
Te rog sa schimbi query-ul sa folosesti GROUP BY si HAVING, in loc de selectul ala intern de la WHERE (vezi ca era comentariu la taskul de pe Trello